### ISSUE TITLE
Jest dependencies breaking cdktf init in fresh directory

### ISSUE THREAD

**Issue by d8aninja** [hashicorp/terraform-cdk#1766]
TL;DR: _When in a fresh directory, calling `cdktf init` seems to break immediately on failure to build an acceptable dependency tree for the testing suite jest._

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

```cdktf --version
0.10.2
```

### Affected Resource(s)

- the CDK project itself

### Debug Output

https://gist.github.com/d8aninja/f473f1a6d9a4e4987e4c191ae382f551

### Expected Behavior

Successful initialization of the cdktf project.  

### Actual Behavior

Error output indicating jest dependencies were surfaced.

### Steps to Reproduce

Shown; `cdktf init` is called in a fresh directory.

### Important Factoids

Worth noting that I can still `cdktf deploy` after this error is reported.  

OS Info: `Linux pop-os 5.16.19-76051619-generic #202204081339~1649696161~22.04~091f44b SMP PREEMPT Mon Apr 11 17 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by ronnylt** [comment]
Same issue here:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: learn-cdktf-docker@1.0.0
npm ERR! Found: jest@27.5.1
npm ERR! node_modules/jest
npm ERR!   dev jest@"^27.4.3" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer jest@"^28.0.0" from ts-jest@28.0.1
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

```

While resolving: learn-cdktf-docker@1.0.0
Found: jest@27.5.1
node_modules/jest
  dev jest@"^27.4.3" from the root project

Could not resolve dependency:
peer jest@"^28.0.0" from ts-jest@28.0.1
node_modules/ts-jest
  dev ts-jest@"*" from the root project

Fix the upstream dependency conflict, or retry
this command with --force, or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by ansgarm** [comment]
Hi! Thanks for bringing this to our attention!

It seems that `ts-jest` released a new major version that requires jest v28 and we still have some fixed in place for a regression in the earlier version of jest:
https://github.com/hashicorp/terraform-cdk/blob/91e226c862e9461f3e305f99e7b448367109b000/packages/cdktf-cli/templates/typescript/.hooks.sscaff.js#L15-L16

We need to fix that and release a new version of the cdktf-cli to fix the template.

**Comment by ansgarm** [comment]
For anyone blocked by this, until we release a new version (which we'll do as soon as possible!), I created a [remote template](https://github.com/ansgarm/cdktf-init-template-ts-workaround-1766) containing the [fix](https://github.com/hashicorp/terraform-cdk/pull/1767):
```
cdktf init --template https://github.com/ansgarm/cdktf-init-template-ts-workaround-1766/archive/refs/heads/main.zip
```
Read more about remote templates in the [CDK for Terraform docs](https://www.terraform.io/cdktf/create-and-deploy/remote-templates).

**Comment by ansgarm** [comment]
CDKTF [v0.10.4](https://github.com/hashicorp/terraform-cdk/releases/tag/v0.10.4) is out 🎉

You can now either upgrade to that version or use the workaround template I posted above 🙂
